packages/synapse-react-client/src/components/Plot/PlotsContainer.tsx

```diff
@@ -0,0 +1,323 @@
+import { getFacets } from '../widgets/facet-nav/FacetNav'
+import FacetNavPanel, { PlotType } from '../widgets/facet-nav/FacetNavPanel'
+import { applyChangesToValuesColumn } from '../widgets/query-filter/FacetFilterControls'
+import { useAtomValue } from 'jotai'
+import { Button, Box } from '@mui/material'
+import FacetNav from '../widgets/facet-nav/FacetNav'
+import QueryWrapperSynapsePlot from '../QueryWrapperPlotNav/QueryWrapperSynapsePlot'
+import useFacetPlots from '../widgets/facet-nav/useFacetPlots'
+import useDeepCompareEffect from 'use-deep-compare-effect'
+import { PlotType as PlotlyPlotType } from 'plotly.js-basic-dist'
+
+const DEFAULT_VISIBLE_PLOTS = 2
+type ShowMoreState = 'MORE' | 'LESS' | 'NONE'
+
+export type PlotsContainerProps = {
+  facetsToPlot?: string[]
+  customPlots?: QueryWrapperSynapsePlotProps[]
+}
+type CustomPlotIdentifier = {
+  title: string
+  __custom: true
+}
+export type PlotIdentifier = UniqueFacetIdentifier | CustomPlotIdentifier
+
+export type UiPlotState = {
+  plotId: PlotIdentifier
+  isHidden: boolean
+  plotType: PlotType
+  index?: number
+}
+const plotMatchesDefinition = (
+  definition: PlotIdentifier,
+  plotId: PlotIdentifier,
+) => {
+  if ('__custom' in plotId && '__custom' in definition) {
+    return definition.title == plotId.title
+  } else if (!('__custom' in plotId) && !('__custom' in definition)) {
+    return facetObjectMatchesDefinition(definition, plotId)
+  } else {
+    return false
+  }
+}
+
+function convertFacetNavPlotTypeToPlotlyPlotType(
+  facetNavPlotType: PlotType,
+): PlotlyPlotType {
+  if (facetNavPlotType === 'BAR') {
+    return 'bar'
+  } else if (facetNavPlotType === 'PIE') {
+    return 'pie'
+  }
+  return 'pie'
+}
+
+function convertPlotlyPlotTypeToFacetNavPlotType(
+  plotlyPlotType: PlotlyPlotType,
+): PlotType {
+  if (plotlyPlotType === 'bar') {
+    return 'BAR'
+  } else if (plotlyPlotType === 'pie') {
+    return 'PIE'
+  }
+  return 'PIE'
+}
+
+const DEFAULT_PLOT_TYPE: PlotType = 'PIE'
+
+const DEFAULT_FACETS_TO_PLOT: string[] = []
+const DEFAULT_CUSTOM_PLOTS: QueryWrapperSynapsePlotProps[] = []
+
+export default function PlotsContainer(props: PlotsContainerProps) {
+  const {
+    facetsToPlot = DEFAULT_FACETS_TO_PLOT,
+    customPlots = DEFAULT_CUSTOM_PLOTS,
+  } = props
+  const { getCurrentQueryRequest, executeQueryRequest, error } =
+    useQueryContext()
+  const data = useAtomValue(tableQueryDataAtom)
+  const isLoadingNewBundle = useAtomValue(isLoadingNewBundleAtom)
+  const { showPlotVisualization } = useQueryVisualizationContext()
+  const [plotUiStateArray, setPlotUiStateArray] = useState<UiPlotState[]>([])
+  const facetNavPanelPropsArray = useFacetPlots(facetsToPlot)
+
+  const getPlotType = (plotId: PlotIdentifier): PlotType => {
+    const plotType = plotUiStateArray.find(item =>
+      plotMatchesDefinition(plotId, item.plotId),
+    )?.plotType
+    return plotType ?? 'PIE'
+  }
+
+  useEffect(() => {
```

Comment:
Should consider rewriting the effect so that instead of reinitializing the state array it
1. Checks each plot to see if it exists in the state array. If not, append it to the existing state
2. Checks if any plots in state have been removed from props. If so, remove that plot from the state array.

This could be one hook, or it could be split by plot group (custom, facet), or it could be split by behavior (add plots that were added to props, remove plots that were removed from props)